src/json-api/json-api-hapi-plugin.ts

```diff
@@ -126,29 +127,57 @@ export class JsonApiHapiPlugin {
       },
     }];
 
-    const preview: Hapi.IRouteConfiguration[] = [{
-      method: 'GET',
-      path: '/preview/{projectId}-{deploymentId}',
-      handler: {
-        async: this.getPreviewHandler,
-      },
-      config: {
-        bind: this,
-        auth: openAuth,
-        pre: [
-          this.authorizeOpenDeployment,
-        ],
-        validate: {
-          params: {
-            projectId: Joi.number().required(),
-            deploymentId: Joi.number().required(),
+    const preview: Hapi.IRouteConfiguration[] = [
+      {
+        method: 'GET',
+        path: '/preview/{projectId}-{deploymentOrBranch}/{token}',
```

Comment:
From the UI's point of view, `{projectId}-{deploymentId}` is an opaque value that's just the deployment ID. It would be better if the latest branch deployment information would be received from e.g. a `/preview/{projectId}/{branchId}/{token}` route.